frontend/src/app/my/mentorship/programs/[programKey]/modules/[moduleKey]/issues/[issueId]/page.tsx

```diff
@@ -0,0 +1,509 @@
+
+  return (
+    <div className="min-h-screen bg-white p-8 text-gray-700 dark:bg-[#212529] dark:text-gray-300">
+      <div className="mx-auto max-w-5xl">
+        <div className="mb-6 flex flex-col gap-4 sm:flex-row sm:items-center sm:justify-between">
+          <div className="min-w-0 flex-1">
+            <h1 className="text-2xl font-bold sm:text-3xl">
+              <span className="break-words">{issue.title}</span>
+            </h1>
+            <div className="mt-1 flex items-center gap-2 text-sm text-gray-500">
+              <span>
+                {issue.organizationName}/{issue.repositoryName} • #{issue.number}
+              </span>
+              <span
+                className={`inline-flex items-center rounded-lg px-2 py-0.5 text-xs font-medium ${
+                  issue.state === 'open'
+                    ? 'bg-[#238636] text-white'
+                    : issue.isMerged
+                      ? 'bg-[#8657E5] text-white'
+                      : 'bg-[#DA3633] text-white'
+                }`}
+              >
+                {issue.state === 'open' ? 'Open' : issue.isMerged ? 'Merged' : 'Closed'}
+              </span>
+            </div>
+          </div>
+          <ActionButton url={issue.url} tooltipLabel="View on GitHub">
+            <FontAwesomeIcon icon={faLink} /> View on GitHub
+          </ActionButton>
+        </div>
+
+        <SecondaryCard title={<AnchorTitle title="Description" />}>
+          <div className="prose dark:prose-invert line-clamp-[15] max-w-none">
+            <Markdown content={issue.body || 'No description.'} />
+          </div>
+        </SecondaryCard>
+
+        <SecondaryCard title={<AnchorTitle title="Task Timeline" />}>
+          <div className="space-y-4 text-sm text-gray-700 dark:text-gray-300">
+            <div>
+              <span className="font-medium">Assigned:</span>{' '}
+              {data?.getModule?.taskAssignedAt
+                ? new Date(data.getModule.taskAssignedAt).toLocaleDateString()
+                : 'Not assigned'}
+            </div>
+
+            <div className="flex flex-col gap-2">
+              <div className="flex flex-wrap items-center">
+                <span className="font-medium">Deadline: </span>
+                {isEditingDeadline && canEditDeadline ? (
+                  <input
+                    type="date"
+                    value={deadlineInput}
+                    onChange={async (e) => {
+                      const newValue = e.target.value
+                      setDeadlineInput(newValue)
+
+                      if (newValue && !settingDeadline && issueId) {
+                        const [year, month, day] = newValue.split('-').map(Number)
+                        const utcEndOfDay = new Date(
+                          Date.UTC(year, month - 1, day, 23, 59, 59, 999)
+                        )
+                        const iso = utcEndOfDay.toISOString()
+
+                        await setTaskDeadlineMutation({
```

Comment:
There's no way to clear up the deadline 🤔 

If I click into an existing deadline and click `Clean` it does remove the date, but there's no way to save that state. I think clicking `Clear` should clean the date and automatically save it.